# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmJS.h
# Commit: 7805ddfe2bfa
# Full Hash: 7805ddfe2bfa83383dc422ed1e8ac70bda627a89
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-19 10:01:16
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Rename TypeCode/ValType/RefType::Any to Extern. r=lth
#   
#   This commit renames the TypeCode/ValType/RefType occurrences of 'any' to 'extern'
#   to match what they actually represent. All public API facing occurrences have
#   already been switched to 'externref' so this should only be aligning the
# ==============================================================================

diff -r 4fa52a0617de -r 7805ddfe2bfa js/src/wasm/WasmJS.h
--- a/js/src/wasm/WasmJS.h	Tue Aug 18 16:58:21 2020 +0000
+++ b/js/src/wasm/WasmJS.h	Tue Aug 18 16:58:32 2020 +0000
@@ -103,7 +103,7 @@
 // optimizing compiler tier.
 bool CodeCachingAvailable(JSContext* cx);
 
-// General reference types (anyref, funcref) and operations on them.
+// General reference types (externref, funcref) and operations on them.
 bool ReftypesAvailable(JSContext* cx);
 
 // Experimental (ref T) types and structure types.
@@ -168,10 +168,10 @@
 bool IsSharedWasmMemoryObject(JSObject* obj);
 
 // Check a value against the given reference type kind.  If the targetTypeKind
-// is RefType::Any then the test always passes, but the value may be boxed.  If
-// the test passes then the value is stored either in fnval (for RefType::Func)
-// or in refval (for other types); this split is not strictly necessary but is
-// convenient for the users of this function.
+// is RefType::Extern then the test always passes, but the value may be boxed.
+// If the test passes then the value is stored either in fnval (for
+// RefType::Func) or in refval (for other types); this split is not strictly
+// necessary but is convenient for the users of this function.
 //
 // This can return false if the type check fails, or if a boxing into AnyRef
 // throws an OOM.